src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,203 @@
+                } else if (arr[0].equals("deadline")) {
+                    parseDeadline(arr);
+                } else if (arr[0].equals("event")) {
+                    parseEvent(arr);
+                } else if (input.equals("list")) {
+                    parseList();
+                } else if (arr[0].equals("find")) {
+                    parseFind(arr);
+                } else {
+                    throw new InvalidCommandException("Command not Found");
+                }
+                storage.saveFile(taskList);
+            } catch (InvalidCommandException e) {
+                System.out.println(e.toString());
+            } catch (EmptyDescriptionException e) {
+                System.out.println(e.toString());
+            } catch (InvalidValueException e) {
+                System.out.println(e.toString());
+            } catch (DateTimeParseException e) {
+                System.out.println(e.getMessage());
+            }
+        } while (!input.equals("bye"));
+    }
+
+    /**
+     * Deals with the user input when the user types "done".
+     *
+     * @param arr Array of strings from the user input.
+     * @throws InvalidCommandException If the following input is not valid.
+     */
+    public void parseDone(String[] arr) throws InvalidCommandException {
+        if (arr.length == 1) {
+            throw new InvalidCommandException("Please specify a number");
+        } else if (!isNumeric(arr[1])) {
+            throw new InvalidCommandException("Please enter a number");
+        } else if (taskList.getSize() == 0) {
+            throw new InvalidCommandException("You have not added any task!");
+        } else if ((Integer.parseInt(arr[1]) > taskList.getSize()
+                || Integer.parseInt(arr[1]) <= 0)) {
+            throw new InvalidValueException("Enter a valid number!");
+        } else {
+            ui.printDone();
+            taskList.markAsDone(parseInt(arr[1]) - 1);
+            ui.printCurrentTask(taskList, parseInt(arr[1]) - 1);
+        }
+    }
+
+    /**
+     * Deals with the user input when the user types "delete".
+     *
+     * @param arr Array of strings from the user input.
+     * @throws InvalidCommandException If the following input is not valid.
+     * @throws InvalidValueException If the following input is not a valid number.
+     */
+    public void parseDelete(String[] arr) throws InvalidCommandException,
+            InvalidValueException {
+        if (arr.length == 1) {
+            throw new InvalidCommandException("Please specify a number");
+        } else if (!isNumeric(arr[1])) {
+            throw new InvalidCommandException("Please enter a number");
+        } else if (taskList.getSize() == 0) {
+            throw new InvalidCommandException("You have not added any task!");
+        } else if ((Integer.parseInt(arr[1]) > taskList.getSize()
+                || Integer.parseInt(arr[1]) <= 0)) {
+            throw new InvalidValueException("Enter a valid number!");
```

Comment:
code here is duplicated from the `parseDone(String[] arr)` method.  you can consider creating  a
method to extract and validate the index